src/main/java/io/aiven/kafka/auth/AivenAclAuthorizer.java

```diff
@@ -80,41 +73,56 @@ public void configure(final java.util.Map<String, ?> configs) {
 
         configFile = config.getConfigFile();
         jsonReader = new AclJsonReader(configFile.toPath());
-        checkAndUpdateConfig();
+        reloadAcls();
+        final WatchKey watchKey = subscribeToAclChanges(configFile);
+        scheduledExecutorService.scheduleWithFixedDelay(() -> {
```

Comment:
BTW, @willyborankin . Wouldn't it make sense to use a polling loop with a block call instead of a scheduled task with regular execution? Something like:

```java
while (true) {
    WatchKey key = watcher.take();  // block until an event happens
}
```

Reply:
Nope better to use a scheduler, since in such a loop you need to add a timer to reduce CPU utilization

Reply:
> since in such a loop you need to add a timer to reduce CPU utilization
By timer you mean a sleep of sorts? I'm not really suggesting a hot loop, `take` is blocking.

Reply:
Augusto's got a point. https://www.baeldung.com/java-nio2-watchservice gives an example of how to use `take` in a loop.

Reply:
`take` does not seem to work because in case of WatchKey corruption(e.g. folder deletion or something else) we need to re-register the WatchService. And we should either guarantee that we will wait until WatchService will be re-registered(e.g. using one more watcher or cycle or whatever) in the same thread before `take` or otherwise we will have to have one more thread for that. Both approaches IMO complicate the things. But if you guys think that's a better option - I can implement any of them.

Reply:
If it gets too painful, just interval check of the file's modification timestamp (like it's now) will suffice

Reply:
@AnatolyPopov just to see if I got it. The problem happens when:

1. The watch service is created
2. The file is registered to the service
3. The call to `watchService.take();` blocks
4. The folder is deleted

The thread on `3` is not notified, even thought the complete folder is gone, and the thread is stuck for ever? So the solution would be to have a separate thread pooling the file system to see if the file is gone, to create a new service, and register a new watcher?

Reply:
Not exactly.
1. The watch service is created
2. The file is registered to the service
3. The WatchKey is corrupted(for example the folder is deleted or another reason, e.g. `WatchKey#reset` returns false)
4. The watch service needs to be re-registered but the registration failed.
5. On the next iteration the call to watchService.take(); blocks

We can not fail completely on `4` because we need to retry at some point or perhaps need to retry indefinitely, so exception should be just logged and not rethrown.

Current version does not suffer from the same issue. We just re-register the watch service if something goes wrong and never block.